app/services/ams/backfill_asset_validation_status.rb

```diff
@@ -0,0 +1,164 @@
+    end
+
+    def fresh_run
+      write_asset_ids_to_file
+      FileUtils.rm(PROCESSED_IDS_PATH) if File.exist?(PROCESSED_IDS_PATH)
+
+      run(ids_file: ALL_IDS_PATH)
+    end
+
+    def resume
+      msg = 'Run #fresh_run before attempting to resume'
+      raise StandardError, msg unless File.exist?(ALL_IDS_PATH) && File.exist?(PROCESSED_IDS_PATH)
+
+      setup_remaining_ids_file
+
+      run(ids_file: REMAINING_IDS_PATH)
+    end
+
+    def run(ids_file:)
+      @ids ||= File.read(ids_file).split("\n")
+      progressbar = ProgressBar.create(total: ids.size, format: '%a %e %P% Processed: %c from %C')
+
+      # Use #begin here to avoid the need to repeatedly open and close the processed_file each time
+      # we need to write to it. The #ensure makes sure the file closes properly even if an error arises,
+      # preventing any data loss. In addition, it conserves IO processing resources by not continuously
+      # opening and closing the file.
+      begin
+        # Suppress most ActiveRecord logging to be able to clearly see the ProgressBar's progress
+        original_log_level = ActiveRecord::Base.logger.level
+        ActiveRecord::Base.logger.level = Logger::ERROR
+
+        processed_file = File.open(PROCESSED_IDS_PATH, 'a')
+        ids.each do |id|
+          # This nested #begin lets us log the `id` currently being processed if an error is thrown
+          begin # rubocop:disable Style/RedundantBegin
+            logger.info("Starting ID: #{id}")
+            backfill_validation_status(id)
+            processed_file.puts(id)
+            progressbar.increment
+          rescue => e
+            logger.error("#{e.class} | #{e.message} | #{id} | Continuing...")
+          end
+        end
+      ensure
+        ActiveRecord::Base.logger.level = original_log_level
+        processed_file&.close
+      end
+    end
+
+    private
+
+    def backfill_validation_status(id)
+      asset = Asset.find(id)
+      attrs_for_actor = {}
+      raise StandardError, "Unable to find admin data for Asset #{asset.id}" if asset.admin_data.blank?
+
+      raw_source_data = if asset.admin_data.bulkrax_importer_id.present?
+                          raw_data_from_bulkrax_entry(asset.admin_data.bulkrax_importer_id, asset)
+                        elsif asset.admin_data.hyrax_batch_ingest_batch_id.present?
+                          raw_data_from_batch_item(asset.admin_data.hyrax_batch_ingest_batch_id, asset)
+                        else
+                          raise StandardError, "Unable to find source data for Asset #{asset.id}"
+                        end
+
+      parsed_source_data = AAPB::BatchIngest::PBCoreXMLMapper.new(raw_source_data).asset_attributes
+      attrs_for_actor['intended_children_count'] = parsed_source_data[:intended_children_count]
+      if attrs_for_actor['intended_children_count'].blank?
+        raise StandardError, "Unable to count intended children for Asset #{asset.id}"
+      end
+
```

Comment:
if you havent loaded the asset here - you can create a background job to do the actor stack and save and never load the asset in this loop.

Reply:
sidebar: what if there was an activerecord object called "CleanupRun" that had id, item_id, item_type, job_name, status, error columns.